test/test_runner_battle.c

```diff
@@ -1151,35 +1156,31 @@ static s32 TryMessage(s32 i, s32 n, const u8 *string)
 
         event = &DATA.queuedEvents[i].as.message;
         // Test_MgbaPrintf("Looking for: %S Found: %S\n", event->pattern, string); // Useful for debugging.
-        for (j = k = 0; ; j++, k++)
+        j = 0; k = 0;
+        while (1)
         {
-            if (event->pattern[k] == CHAR_SPACE)
-            {
-                switch (string[j])
-                {
-                case CHAR_SPACE:
-                case CHAR_PROMPT_SCROLL:
-                case CHAR_PROMPT_CLEAR:
-                case CHAR_NEWLINE:
-                    j++;
-                    k++;
-                    break;
-                }
+            if (IS_SKIPOVER_CHAR(event->pattern[k])) {
+                k++;
+                continue;
             }
-            if (event->pattern[k] == EOS)
-            {
-                // Consume any trailing '\p'.
-                if (string[j] == CHAR_PROMPT_CLEAR)
-                    j++;
+            if (IS_SKIPOVER_CHAR(string[j])) {
+                j++;
+                continue;
+            }
+            if (string[j] == EXT_CTRL_CODE_BEGIN && string[j+1] <= EXT_CTRL_CODE_COLOR_HIGHLIGHT_SHADOW) {
+                j += 3;
+                continue;
             }
             if (string[j] != event->pattern[k])
             {
                 break;
             }
-            else if (string[j] == EOS)
+            if (string[j] == EOS || event->pattern[k] == EOS)
```

Comment:
I think `event->pattern[k] == EOS` cannot occur, because the previous line is `if (string[j] != event->pattern[k]) break;`, so if `string[k] != EOS && event->pattern[k] == EOS` we should have already `break`ed?

If it could occur, I think this code would be wrong because we shouldn't exit this code with a success if we get to the end of `event->pattern` without also getting to the end of `string`.